Model/DispatchPlugin.php

```diff
@@ -57,17 +57,18 @@ public function aroundDispatch(
         }
 
         try {
-            if ($messageTimestamp !== false &&
-                $lastModificationTime = $this->modificationTimeRepo->getLastModificationTime($messageGroupId)
-            ) {
-                if (!$this->isUnmodifiedSince($lastModificationTime, $messageTimestamp)) {
+            $lastModification = $this->modificationTimeRepo->getLastModification($messageGroupId);
+            $lastModificationTime = $lastModification['timestamp'] ?? null;
+            $lastVersion = $lastModification['version'] ?? null;
+
+            if ($messageTimestamp !== false) {
+                if ($lastModificationTime !== null && $messageTimestamp < $lastModificationTime) {
                     $this->response->setStatusCode(412);
                     return $this->response;
                 }
-                $updateTimestamp = $messageTimestamp;
+                $currentTimestamp = $messageTimestamp;
             } else {
-                $lastModificationTime = null;
-                $updateTimestamp = \time();
+                $currentTimestamp = \time();
```

Comment:
I wonder what use the default timestamp is? This feature is pretty pointless if the client doesn't specify a timestamp. Perhaps we should only invoke the feature if _both_ the message group ID _and_ a timestamp are provided?